pool.go

```diff
@@ -253,12 +253,17 @@ func (p *Pool) retrieveWorker() (w *goWorker) {
 
 // revertWorker puts a worker back into free pool, recycling the goroutines.
 func (p *Pool) revertWorker(worker *goWorker) bool {
-	if capacity := p.Cap(); (capacity > 0 && p.Running() > capacity) || atomic.LoadInt32(&p.state) == CLOSED {
+	if capacity := p.Cap(); capacity > 0 && p.Running() > capacity {
```

Comment:
You shouldn't remove `atomic.LoadInt32(&p.state) == CLOSED` in this precheck, this condition is still useful for common cases, just add a double check in the lock scope.

Reply:
Bro, I got it.